pkg/cache/cache.go

```diff
@@ -721,6 +725,20 @@ func (c *Cache) MatchingClusterQueues(nsLabels map[string]string) sets.Set[strin
 	return cqs
 }
 
+// ShouldCheckWorkloadsPreemption returns true if it has workloads pending preemption now.
+func (c *Cache) ShouldCheckWorkloadsPreemption() bool {
+	c.RLock()
+	defer c.RUnlock()
+	// Depending on the overall state this may end up iterating over all the workloads present
+	// in the cache (workloads having QuotaReservation).
+	for _, cq := range c.clusterQueues {
+		if now, _ := cq.GetPreemptingWorkloads(c.admissionChecks); len(now) > 0 {
```

Comment:
```suggestion
		if cq.HasPreemptingWorkloads(c.admissionChecks) {
```

Even if the logic looks similar, returning true/false would be much faster.

Reply:
same as L729

Reply:
ok, but make it return true/false, instead of a list that is not used.